scripts/template_upload_rhevm.py

```diff
@@ -240,30 +245,74 @@ def templatize_vm(api, template_name, cluster):
+            edomain_conn.get_address())
+
+
+def cleanup_empty_dir_on_edomain(path, edomainip, sshname, sshpass):
+    """Cleanup all the empty directories on the edomain/edomain_id/master/vms
+    else api calls will result in 400 Error with ovf not found,
 
     Args:
         api: API to chosen RHEVM provider.
         edomain: Export domain of chosen RHEVM provider.
+        edomainip: edomainip to connect through ssh.
+        sshname: edomain ssh credentials.
+        sshpass: edomain ssh credentials.
     """
-    command = 'rm %s' % ovaname
+    print "RHEVM: Deleting the empty directories on edomain/vms file..."
+    ssh_client = make_ssh_client(edomainip, sshname, sshpass)
+    command = 'cd {}/master/vms && find . -maxdepth 1 -type d -empty -delete'.format(path)
     exit_status, output = ssh_client.run_command(command)
+    if exit_status != 0:
+        print "RHEVM: Error while deleting the empty directories on path.."
+        print output
 
-    temporary_vm = api.vms.get(TEMP_VM_NAME)
-    if temporary_vm is not None:
-        temporary_vm.delete()
-
-    temporary_template = api.templates.get(TEMP_TMP_NAME)
-    if temporary_template is not None:
-        temporary_template.delete()
 
-    # waiting for template on export domain
-    wait_for(check_edomain_template, [api, edomain], fail_condition=False, delay=5)
+def cleanup(api, edomain, ssh_client, ovaname):
+    """Cleans up all the mess that the previous functions left behind.
 
-    unimported_template = api.storagedomains.get(edomain).templates.get(TEMP_TMP_NAME)
-    if unimported_template is not None:
-        unimported_template.delete()
+    Args:
+        api: API to chosen RHEVM provider.
+        edomain: Export domain of chosen RHEVM provider.
+    """
+    try:
+        print "RHEVM: Deleting the  .ova file..."
+        command = 'rm %s' % ovaname
+        exit_status, output = ssh_client.run_command(command)
+
+        print "RHEVM: Deleting the temp_vm on sdomain..."
+        temporary_vm = api.vms.get(TEMP_VM_NAME)
+        if temporary_vm:
+            temporary_vm.delete()
+
+        print "RHEVM: Deleting the temp_template on sdomain..."
+        temporary_template = api.templates.get(TEMP_TMP_NAME)
+        if temporary_template:
+            temporary_template.delete()
+
+        # waiting for template on export domain
+        unimported_template = api.storagedomains.get(edomain).templates.get(
+            TEMP_TMP_NAME)
+        print "RHEVM: waiting for template on export domain..."
+        wait_for(check_edomain_template, [unimported_template],
+                 fail_condition=False, delay=5)
+
+        if unimported_template:
+            print "RHEVM: deleting the template on export domain..."
+            unimported_template.delete()
+
+        print "RHEVM: waiting for template delete on export domain..."
+        wait_for(is_edomain_template_deleted, [unimported_template],
+                 fail_condition=False, delay=5)
+
```

Comment:
thos-at-redhat: the edomain state is changed to maintenance state and verified, before and after cleanup.